### PR TITLE
fix Execution order of ci_check_unittest, and add it to Linux_py35, t…

### PR DESCRIPTION
**Pull request by zhwesky2010**
- fix Execution order of ci_check_unittest, and add it to Mac_ci_Py35, Linux_ci, Linux_py35.

**Execution order:**
1. check approve, if approved by zhouwei25 (recommended) or luotao1 or XiaoguangHu01 or phlrain or lanxianghit, pass CI directly.
2. cmake first time for building.
3. building
4. run unit test
5. cmake second time in develop of upstream
6. check whether the unit test have been changed, if reduce, exit with approval error.

for example, unit_test is deleted in this PR #21673 , It will be intercepted.

image

image

The wiki describing the rules is [WIKI](https://github.com/PaddlePaddle/Paddle/wiki/Deleting-unit-test-is-forbidden)

### CODE REVIEW

File: paddle/scripts/paddle_build.sh [hunk 1]
```diff
@@ -1168,18 +1179,22 @@ function main() {
...
       macbuild)
         cmake_gen ${PYTHON_ABI:-""}
         build_mac
         ;;
       cicheck_py35)
+        check_approvals_of_unittest 1
         cmake_gen ${PYTHON_ABI:-""}
         build ${parallel_number}
         parallel_test
+        check_change_of_unittest ${PYTHON_ABI:-""}
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh]: cicheck和windows里也需要加，cicheck和ckcheck_py35是有挺多区别的。

Comment (zhwesky2010) [reply paddle/scripts/paddle_build.sh]: 好的，cicheck也加上，windows走不了这套，是在ci里自行写bat编译的，没有运行paddle_build.sh

Comment (luotao1) [reply paddle/scripts/paddle_build.sh]: 那windows后续准备用什么办法监控呢？很多单测在windows上面是disable了。

Comment (zhwesky2010) [reply paddle/scripts/paddle_build.sh]: Windows要么重新通过新的脚本来做一套，主要是.sh不能复用，.py可以复用，或者看CI机器能否部署什么工具可以模拟跑.sh，这个在mac/linux弄好了我再针对Windows测一下

Comment (luotao1) [reply paddle/scripts/paddle_build.sh]: 好的。

**Review by luotao1 — COMMENTED** [review]
1. 建议不要覆盖之前的commit，这样reviewer看不到commit之间的变化
2. 报错的时候会提示删了哪个单测么？

**Comment by zhwesky2010** [comment]
> 1. 建议不要覆盖之前的commit，这样reviewer看不到commit之间的变化
> 2. 报错的时候会提示删了哪个单测么？

会提示删了哪个单测

File: paddle/scripts/paddle_build.sh [hunk 2]
```diff
@@ -511,54 +512,64 @@ function generate_api_spec() {
...
+        if [ "$unittest_spec_diff" != "" ]; then
+            APPROVALS=`echo ${approval_line}|python ${PADDLE_ROOT}/tools/check_pr_approval.py 1 46782768 6836917 43953930 47554610`
+            echo "current pr ${GIT_PR_ID} got approvals: ${APPROVALS}"
+            if [ "${APPROVALS}" == "FALSE" ]; then
+                echo "****************"
+                echo -e "You must have one RD (zhouwei25 (recommended) or luotao1 or XiaoguangHu01 or phlrain or lanxianghit) approval for the deletion of unit tests.\n"
+                echo "There are one approved errors."
+                echo "****************"
+                exit 1
+            fi
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh:542]: - 错误截图里，每行命令可以不打出来。
- 最终出错的单测可以放到535行后面。
- 535行，需要加一个wiki说明下 该规范的背景/流程，末尾加文字“以下是删去的单测。”

**Comment by luotao1** [comment]
- 黄色的执行语句可以不打出来么？
- 报错语句可否整理如下？

It is forbidden to disable or delete the unit-tests. You must have one RD (zhouwei25 (recommended) or luotao1 or XiaoguangHu01 or phlrain or lanxianghit) approval for the deletion of unit tests. If you have any problems with deleting the unit test, please read the specification [https://github.com/PaddlePaddle/Paddle/wiki/OP-Specification:-Deleting-unit-test-is-forbidden]. Following unit-tests are deleted in this PR:

-  assign_op_test
-  gather_test
-  scatter_test
-  beam_search_decode_op_test

**Comment by zhwesky2010** [comment]
> * 黄色的执行语句可以不打出来么？
> * 报错语句可否整理如下？
> 
> It is forbidden to disable or delete the unit-tests. You must have one RD (zhouwei25 (recommended) or luotao1 or XiaoguangHu01 or phlrain or lanxianghit) approval for the deletion of unit tests. If you have any problems with deleting the unit test, please read the specification [https://github.com/PaddlePaddle/Paddle/wiki/OP-Specification:-Deleting-unit-test-is-forbidden]. Following unit-tests are deleted in this PR:
> 
> * assign_op_test
> * gather_test
> * scatter_test
> * beam_search_decode_op_test
好的

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**